internal/executor/contracts/proposal_strategy.go

```diff
@@ -0,0 +1,273 @@
+	if err := checkProposalType(ProposalType(pt)); err != nil {
+		return boltvm.Error(boltvm.GovernanceIllegalProposalTypeCode, fmt.Sprintf(string(boltvm.GovernanceIllegalProposalTypeMsg), pt))
+	}
+
+	ps := &ProposalStrategy{}
+	if !g.GetObject(ProposalStrategyKey(pt), ps) {
+		return boltvm.Error(boltvm.GovernanceNonexistentProposalStrategyCode, fmt.Sprintf(string(boltvm.GovernanceNonexistentProposalStrategyMsg), pt))
+	}
+
+	pData, err := json.Marshal(ps)
+	if err != nil {
+		return boltvm.Error(boltvm.GovernanceInternalErrCode, fmt.Sprintf(string(boltvm.GovernanceInternalErrMsg), err.Error()))
+	}
+	return boltvm.Success(pData)
+}
+
+func (g *GovStrategy) changeStatus(pt string, trigger, lastStatus string) (bool, []byte) {
+	ps := &ProposalStrategy{}
+	if ok := g.GetObject(ProposalStrategyKey(pt), ps); !ok {
+		return false, []byte("this proposal strategy does not exist")
+	}
+
+	ps.setFSM(governance.GovernanceStatus(lastStatus))
+	err := ps.FSM.Event(trigger)
+	if err != nil {
+		return false, []byte(fmt.Sprintf("change status error: %v", err))
+	}
+
+	g.SetObject(ProposalStrategyKey(pt), *ps)
+	return true, nil
+}
+
+func ProposalStrategyKey(ps string) string {
+	return fmt.Sprintf("%s-%s", PROPOSALSTRATEGY_PREFIX, ps)
+}
+
+func (g *GovStrategy) checkPermission(permissions []string, regulatedAddr, regulatorAddr string, specificAddrsData []byte) error {
+	for _, permission := range permissions {
+		switch permission {
+		case string(PermissionSelf):
+			if regulatedAddr == regulatorAddr {
+				return nil
+			}
+		case string(PermissionAdmin):
+			res := g.CrossInvoke(constant.RoleContractAddr.Address().String(), "IsAnyAvailableAdmin",
+				pb.String(regulatorAddr),
+				pb.String(string(GovernanceAdmin)))
+			if !res.Ok {
+				return fmt.Errorf("cross invoke IsAvailableGovernanceAdmin error:%s", string(res.Result))
+			}
+			if "true" == string(res.Result) {
+				return nil
+			}
+		case string(PermissionSpecific):
+			specificAddrs := []string{}
+			if err := json.Unmarshal(specificAddrsData, &specificAddrs); err != nil {
+				return err
+			}
+			for _, addr := range specificAddrs {
+				if addr == regulatorAddr {
+					return nil
+				}
+			}
+		default:
+			return fmt.Errorf("unsupport permission: %s", permission)
+		}
+	}
+
+	return fmt.Errorf("regulatorAddr(%s) does not have the permission", regulatorAddr)
+}
+
+// Check info =============================================================
+func checkProposalType(pt ProposalType) error {
+	if pt != AppchainMgr && pt != RuleMgr && pt != NodeMgr && pt != RoleMgr && pt != DappMgr && pt != ServiceMgr && pt != ProposalStrategyMgr {
+		return fmt.Errorf("illegal proposal type")
+	}
+	return nil
+}
+
+func CheckProposalStatus(ps ProposalStatus) error {
+	if ps != PROPOSED &&
+		ps != APPROVED &&
+		ps != REJECTED &&
+		ps != PAUSED {
+		return fmt.Errorf("illegal proposal status")
+	}
+	return nil
+}
+
+func CheckStrategyInfo(ps *ProposalStrategy) error {
```

Comment:
Check whether the threshold of ZeroPermission is 0
Check whether the SimpleMajority threshold is in the range (0,1)